### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by Horoshaya**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+
+import styles from './App.css';
+import Header from './components/Header/Header';
+import Main from './components/Main/Main';
+import Footer from './components/Footer/Footer';
+import store from './store/initialState';
+
+// console.log('store', store);
 
 class App extends Component {
```
Comment (GlebShaucou) [inline src/App.js]: You can also change App to a functional component.

File: src/components/DeleteModal/DeleteModal.js [hunk 2]
```diff
@@ -1,16 +1,26 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
 
 import Modal from '../Modal/Modal';
 import styles from './DeleteModal.css';
+import { deletetMovie } from '../../store/actions/actionCreators';
```
Comment (GlebShaucou) [inline src/components/DeleteModal/DeleteModal.js:6]: deletet

File: src/components/GenreTab/GenreTab.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React, { useContext } from 'react';
+import styles from './GenreTab.css';
+import { MainContext } from '../Main/Main';
+
+const GenreTab = ({ isActive, children }) => {
+  const { filterByGenreHandle } = useContext(MainContext);
+  return (
+    <button
+      onClick={() => filterByGenreHandle(children)}
+      className={`${styles.button} ${isActive ? styles.active : ''}`}
```
Comment (GlebShaucou) [inline src/components/GenreTab/GenreTab.js:10]: Useful package for className https://www.npmjs.com/package/classnames

File: src/store/actions/actionCreators.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    if (genre !== 'All') {
+      updatedMovies = allMovies.filter((movie) => movie.genres.includes(genre));
+    }
+
+    dispatch({ type: `${ACTIONS.FILTER_BY_GENRE}`, payload: updatedMovies });
+  };
+};
+
+export const sortByDateAndRating = (type, { allMovies }) => {
+  return (dispatch) => {
```
Comment (GlebShaucou) [inline src/store/actions/actionCreators.js:29]: This could be rewritten as a reducer itself.

File: src/store/actions/actionCreators.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        dispatch({ type: `${ACTIONS.GET_MOVIE}`, payload: movieData.data });
+      })
+      .catch((err) => {
+        console.error("Server doesn't response", err);
+      });
+  };
+};
+
+export const filterByGenre = (genre, { allMovies }) => {
+  return (dispatch) => {
```
Comment (GlebShaucou) [inline src/store/actions/actionCreators.js:17]: filterByGenre could be a reducer function, your action should deliver only `type` and how it should be filered.

File: src/components/GenreTab/GenreTab.js (hunk 3 above)
Comment (Horoshaya) [reply src/components/GenreTab/GenreTab.js:10]: thanks!

File: src/App.js (hunk 1 above)
Comment (Horoshaya) [reply src/App.js]: done

File: src/store/actions/actionCreators.js (hunk 5 above)
Comment (Horoshaya) [reply src/store/actions/actionCreators.js:17]: will be fixed later

**Pull request merged**